crates/services/p2p/src/p2p_service.rs

```diff
@@ -567,54 +557,36 @@ impl<Codec: NetworkCodec> FuelP2PService<Codec> {
                     request_id,
                     response,
                 } => {
-                    match (
-                        self.outbound_requests_table.remove(&request_id),
-                        self.network_codec.convert_to_response(&response),
-                    ) {
+                    let Some(channel) = self.outbound_requests_table.remove(&request_id)
+                    else {
+                        debug!("Send channel not found for {:?}", request_id);
+                        return None;
+                    };
+
+                    let send_ok = match (channel, response) {
                         (
-                            Some(ResponseChannelItem::Block(channel)),
-                            Ok(ResponseMessage::SealedBlock(block)),
-                        ) => {
-                            if channel.send(*block).is_err() {
-                                tracing::error!(
-                                    "Failed to send through the channel for {:?}",
-                                    request_id
-                                );
-                            }
-                        }
+                            ResponseChannelItem::Block(channel),
+                            ResponseMessage::Block(block),
+                        ) => channel.send(block.map(|b| Arc::into_inner(b).expect("There are not other references, we just received this from the network"))).is_ok(),
```

Comment:
Do we even need to use `Arc` for block headers and transactions? Maybe we can  work directly with types